### PR TITLE
feat: add main menu screen from Node Essentials

### PR DESCRIPTION
**Pull request by NathanLovato**
This adds the working title screen to the project. @bramreth here are the things needed right now:

- Loading only scenes in the Demos/ folder. It seems the code currently loads recursively. See DemoList.gd, the file reading happens there.
- Ensuring all demos load and look as intended. Some 3D demos load and look fine, and some are missing content only when loading through this main menu.
- Moving strings to some CSV file and using Godot's translation system like Node Essentials. This may be best in a subsequent PR to avoid making this too big.
- New title graphics (I'll provide this)

Low-priority improvements to do later: 

- Adding an icon per shader category. We need to settle down on the categories and get someone to make polished icons.

### CODE REVIEW

**Comment by bramreth** [comment]
Demos no longer load recursively, and every demo screen has been updated to work in the demo viewer.

I'm going to dive straight into moving strings over to csv in this pr as the extant work hasn't been that much.

**Comment by bramreth** [comment]
@NathanLovato this should be all good to merge now, all that needs doing is:
- New title graphics
- adding group icons and enabling group filters from menu
- adding help instructions
- adding controls to demo scenes (this is only ever, ui_accept to activate the effect, or hover over objects to activate)
- adding a reset button
I think all of these can go in a different PR though

**Review by Arnklit — COMMENTED** [review]
Hi Bram, had a look through the changes and everything looked good to me, just saw some minor things like more vars that should be considered pseudo private.

File: godot/Demos/Blur/Ship.gd [hunk 1]
```diff
@@ -11,16 +11,16 @@ var speed := 0.0
 onready var blur_provider := get_node(blur_provider_path)
 onready var trail_direction := Vector2.UP
 onready var shaking_camera_background := get_node(shaking_camera_background_path)
```
[the remark this answers is not in the document]
Comment (NathanLovato) [reply godot/Demos/Blur/Ship.gd:13]: Yes definitely

File: godot/Demos/SphereMask/LanternPositionToMaterial.gd [hunk 2]
```diff
@@ -1,5 +1,9 @@
 extends Node
 tool
```
[the remark this answers is not in the document]
Comment (NathanLovato) [reply godot/Demos/SphereMask/LanternPositionToMaterial.gd:2]: Yeah in the gdscript styleguide it should be at the top.

**Comment by bramreth** [comment]
> Hi Bram, had a look through the changes and everything looked good to me, just saw some minor things like more vars that should be considered pseudo private.

All these changes are now included :smiley:

**Pull request merged**